Sources/HummingbirdCore/Request/RequestBody+inboundClose.swift

```diff
@@ -0,0 +1,142 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Hummingbird server framework project
+//
+// Copyright (c) 2024 the Hummingbird authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See hummingbird/CONTRIBUTORS.txt for the list of Hummingbird authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import HTTPTypes
+import NIOConcurrencyHelpers
+import NIOCore
+import NIOHTTPTypes
+
+#if compiler(>=6.0)
+@available(macOS 15, iOS 18, tvOS 18, *)
+extension RequestBody {
+    /// Run provided closure but cancel it if the inbound request part stream is closed.
+    ///
+    /// This function is designed for use with long running requests like server sent events. It assumes you
+    /// are not going to be using the request body after calling as it consumes the request body, it also assumes
+    /// you havent edited the request body prior to calling this function.
+    ///
+    /// If the response finishes the connection will be closed.
+    ///
+    /// - Parameters
+    ///   - operation: The actual operation
+    ///   = onInboundClose: handler invoked when inbound is closed
+    /// - Returns: Return value of operation
+    @available(macOS 15, iOS 18, tvOS 18, *)
+    public func consumeWithInboundCloseHandler<Value: Sendable>(
```

Comment:
```suggestion
    public func readWithInboundCloseHandler<Value: Sendable>(
```

I think _read_ makes more sense, as `consume` could be interpreted differently